### ISSUE TITLE
Add RPC Request timeout and prevent request blocking

### ISSUE THREAD

**Issue by yogihardi** [eclipse/paho.golang#108]
**Is your feature request related to a problem? Please describe.**
this issue is related to this PR https://github.com/eclipse/paho.golang/pull/98

> The most significant issue was the rpc.Handler.Request function, in which the go routine could block indefinitely waiting for a response from the server.

and I found that there are missed fixes from the last PR on the following line https://github.com/eclipse/paho.golang/blob/e818d6af9263c1ff1c68749d9e87defb8f51690f/paho/extensions/rpc/rpc.go#L76

**Describe the solution you'd like**

* add context to the Request channel to prevent blocking
* add request timeout

**Comment by hspaay** [comment]
@yogihardi thank you for submitting the patch. Any idea what happened to it?
I'm running into the same issue. Handler.Request locks up when no response is received.
Apparently this is fixed in autopaho's rpc but not in paho's rpc.

**Comment by yogihardi** [comment]
@hspaay yes, you're right the problem is on `Handler.Request`, the channel blocks request with no timeout limit.
the solution is adding `Context.WithTimeout`, you can check here https://github.com/eclipse/paho.golang/blob/1235c124f63fd01560d1380a89fc75dfbeeb0497/paho/extensions/rpc/rpc.go#L92

**Comment by hspaay** [comment]
@yogihardi thanks for the feedback. I copied the rpc.go file and made the change. Works like a charm.
